Patch for FACT-656 facter generates error on Solaris kernel zone due to prtdiag
Upstream bug: https://tickets.puppetlabs.com/browse/FACT-656
--- facter-2.1.0/lib/facter/virtual.rb.orig	2014-08-18 10:37:43.330858038 -0600
+++ facter-2.1.0/lib/facter/virtual.rb	2014-08-18 10:38:02.723874811 -0600
@@ -67,7 +67,7 @@
   setcode do
     next "zone" if Facter::Util::Virtual.zone?
 
-    output = Facter::Core::Execution.exec('prtdiag')
+    output = Facter::Core::Execution.exec('/usr/sbin/prtdiag 2>/dev/null')
     if output
       lines = output.split("\n")
       next "parallels"  if lines.any? {|l| l =~ /Parallels/ }
